com.unity.formats.fbx/Editor/Sources/Recorders/FbxRecorder/FbxRecorder.cs

```diff
@@ -21,54 +19,75 @@ protected override void EndRecording(RecordingSession session)
                 throw new System.ArgumentNullException("session");
             }
 
-            var settings = (FbxRecorderSettings)session.settings;
-
-            foreach (var input in m_Inputs)
+            if (Recording)
```

Comment:
This is error-prone: it's very easy for someone to stick a `return` in the middle of this big block and then the `            base.EndRecording(session);` doesn't happen.

What you can do is make it be:
```
if (Recording)
{
   ActuallyEndRecording(session);
}
base.EndRecording(session);
```
and have all that code in this block be in a separate function.

(Side benefit: it means you avoid reindenting it)